app/models/needs/need.rb

```diff
@@ -4,6 +4,7 @@
 
 class Need < ApplicationRecord
   include Filterable
+  self.inheritance_column = 'category'
```

Comment:
If the column were called `type`, we wouldn't even need this line, as rails assumes STI by default with that field name, apparently.